File: python/Problem002.py
# By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.
def fib(max):
	a,b=1,2
	yield a

	while b<max:
		yield b
		a,b=b,a+b

print sum([x for x in fib(4000000) if x%2==0])

File: c++/problem002.cpp
#include <iostream>

// problem001.cpp
// By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

void next(int &a, int &b) {
  int c = a + b;
  
  a = b;
  b = c;
}

int main() {
	int a = 0, b = 1;
  int sum = 0;

	while ( b < 4000000 ) {
		if ( b % 2 == 0 ) {
			sum += b;
		}

		next(a, b);
	}
  
	std::cout << sum << std::endl;
  return 0;
}
